internal/neofs/neofs.go

```diff
@@ -345,6 +376,52 @@ func (x *NeoFS) CreateObject(ctx context.Context, prm layer.PrmObjectCreate) (oi
 	return writer.GetResult().StoredObjectID(), nil
 }
 
+// PrepareMulptipartHeader implements neofs.NeoFS interface method.
+func (x *NeoFS) PrepareMulptipartHeader(ctx context.Context, header object.Object, metaChecksum hash.Hash, homomorphicChecksum hash.Hash, payloadLength uint64) (*object.Object, error) {
+	header.SetCreationEpoch(x.epochGetter.CurrentEpoch())
+
+	var cs checksum.Checksum
+
+	var csBytes [sha256.Size]byte
+	copy(csBytes[:], metaChecksum.Sum(nil))
+
+	cs.SetSHA256(csBytes)
+	header.SetPayloadChecksum(cs)
+
+	if homomorphicChecksum != nil {
+		var csHomoBytes [tz.Size]byte
+		copy(csHomoBytes[:], homomorphicChecksum.Sum(nil))
+
+		cs.SetTillichZemor(csHomoBytes)
+		header.SetPayloadHomomorphicHash(cs)
+	}
+
+	header.SetPayloadSize(payloadLength)
+
+	id, err := header.CalculateID()
```

Comment:
could be done via `header.SetIDWithSignature(x.signer(ctx))`

Reply:
Done!